### PR TITLE
[API Compatiblity] Add `paddle.compat.nn.Softmax`

### PR DESCRIPTION
**Pull request by fangfangssj**
### PR Category

User Experience

### PR Types

New features

### Description

增加了对齐torch.nn.Softmax的paddle.compat.nn.Softmax

### CODE REVIEW

File: python/paddle/compat/nn/__init__.py [hunk 1]
```diff
@@ -613,6 +616,130 @@ def reset_parameters(self) -> None:
...
+            [[[0.73105854, 0.73105854, 0.73105854, 0.73105854],
+              [0.11920292, 0.11920292, 0.11920292, 0.11920292],
+              [0.73105854, 0.73105854, 0.50000000, 0.50000000]],
+             [[0.26894143, 0.26894143, 0.26894143, 0.26894143],
+              [0.88079703, 0.88079703, 0.88079703, 0.88079703],
+              [0.26894143, 0.26894143, 0.50000000, 0.50000000]]])
+
+    """
+
+    @ForbidKeywordsDecorator(
```
Comment (zhwesky2010) [inline python/paddle/compat/nn/__init__.py:725]: infoflow 2025-11-26 21-08-03

torch.nn.Softmax只需要加两个装饰器，改下别名就行吧。

不涉及不兼容升级，怎么还改到compat里去了

File: python/paddle/compat/nn/__init__.py [hunk 2]
```diff
@@ -441,7 +439,13 @@ class Unfold(nn.Unfold):
     strides: Size2
 
     @ForbidKeywordsDecorator(
-        illegal_keys={"kernel_sizes", "dilations", "paddings", "strides"},
+        illegal_keys={
+            "kernel_sizes",
+            "dilations",
+            "paddings",
+            "strides",
+            "name",
```
Comment (zhwesky2010) [inline python/paddle/compat/nn/__init__.py]: name不用处理，无需关注。

File: python/paddle/compat/nn/__init__.py [hunk 3]
```diff
@@ -613,6 +616,130 @@ def reset_parameters(self) -> None:
...
+              [0.11920292, 0.11920292, 0.11920292, 0.11920292],
+              [0.73105854, 0.73105854, 0.50000000, 0.50000000]],
+             [[0.26894143, 0.26894143, 0.26894143, 0.26894143],
+              [0.88079703, 0.88079703, 0.88079703, 0.88079703],
+              [0.26894143, 0.26894143, 0.50000000, 0.50000000]]])
+
+    """
+
+    @ForbidKeywordsDecorator(
+        illegal_keys={"axis", "name"},
```
Comment (zhwesky2010) [inline python/paddle/compat/nn/__init__.py]: name无需关注

File: python/paddle/compat/nn/__init__.py (hunk 2 above)
Comment (fangfangssj) [reply python/paddle/compat/nn/__init__.py]: done

File: python/paddle/compat/nn/__init__.py (hunk 3 above)
Comment (fangfangssj) [reply python/paddle/compat/nn/__init__.py]: done

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]
